test/index.test.js

```diff
@@ -180,7 +180,30 @@ describe('index', () => {
                 .then(() => assert.fail('Should not get here'))
                 .catch((error) => {
                     assert.calledWith(requestMock, expectedOptions);
-                    assert.match(error.message, 'STATUS CODE 404');
+                    assert.match(error.message, 'Cannot find repository');
+                });
+        });
+
+        it('rejects if status code is not 200 & 404', () => {
+            fakeResponse = {
+                statusCode: 500,
+                body: {
+                    error: {
+                        message: 'INternal Server Error'
+                    }
+                }
+            };
+
+            requestMock.yieldsAsync(null, fakeResponse, fakeResponse.body);
+
+            return scm.parseUrl({
+                checkoutUrl: 'https://batman@bitbucket.org/batman/test.git#mynewbranch',
+                token: 'myAccessToken'
+            })
+                .then(() => assert.fail('Should not get here'))
```

Comment:
I think we went over this. Both ways are fine but some people commented that the other way is hard to read so I switched to this one. Fine with me either way.